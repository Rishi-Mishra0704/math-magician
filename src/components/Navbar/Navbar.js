import React from 'react';
import { Link } from 'react-router-dom';
import classes from '../css/Navbar.module.css';

const Navbar = () => (
  <header>
    <nav className={classes.navbar}>
      <h1>Math Magician</h1>
      <ul className={classes.navList}>
        <li>
          <Link to="/" className={classes.navLink}>
            HomePage
          </Link>
        </li>
        <li>
          <Link to="/calculator" className={classes.navLink}>
            Calculator
          </Link>
        </li>
        <li>
          <Link to="/quotes" className={classes['navLink-last-child']}>
            Quotes
          </Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default Navbar;
