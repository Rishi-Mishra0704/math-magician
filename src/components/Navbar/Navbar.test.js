import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from './Navbar';
import '@testing-library/jest-dom/extend-expect';

describe('Navbar component', () => {
  test('renders Math Magician header', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders HomePage link', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders Calculator link', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders Quotes link', () => {
    const { container } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
