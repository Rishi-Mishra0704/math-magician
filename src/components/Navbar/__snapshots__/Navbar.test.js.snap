// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders Calculator link 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Navbar component renders HomePage link 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Navbar component renders Math Magician header 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Navbar component renders Quotes link 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
